app/controllers/welcome_controller.rb

```diff
@@ -3,9 +3,9 @@
 class WelcomeController < ApplicationController
   def index
     if params[:tag]
-      @tutorials = Tutorial.tagged_with(params[:tag]).paginate(page: params[:page], per_page: 5)
```

Comment:
We can condense this and dry this up a bit. - 

def index
  @tutorials = Tutorial.non_classroom.paginate(page: params[:page], per_page: 5)
  @tutorials.tagged_with(params[:tag]) if params[:tag]
end